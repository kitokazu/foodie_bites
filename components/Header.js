import Image from "next/image";
import appLogo from '../images/mcdonalds_logo.png'

function Header() {
  return (
    <div className="flex justify-between bg-white max-w-6xl">
      {/* Left Side */}
      <div className="relative w-24">
        <Image 
          src={appLogo}
          alt="McDonalds Logo"
          width={200}
          height={100}
          />
      </div>
      {/* Center */}
      <div className="flex space-x-4 text-gray-700">
        <p className="link">Home</p>
      </div>

      {/* Right Side */}
    </div>
  );
}

export default Header