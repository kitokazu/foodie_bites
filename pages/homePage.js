import { Inter } from '@next/font/google';
import Head from 'next/head'
import Header from '@/components/Header'
import Home from '../components/Home';

const inter = Inter({ subsets: ['latin'] })

export default function homePage() {
  return (
    <>
      <Head>
        <title>Home</title>
        <meta name="description" content="Generated by create next app" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Header />
      <Home />
    </>
  )
}
