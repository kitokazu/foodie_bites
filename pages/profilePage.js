import { useState } from 'react'
import { sum as addition } from './math.utils'
import { Inter } from '@next/font/google'
import styles from '@/styles/Home.module.css'
import Head from 'next/head'
import Header from '@/components/Header'
import Profile from '@/components/Profile'
import Modal from '@/components/Modal'

const inter = Inter({ subsets: ['latin'] })

function profilePage({ name, image }) {
  return (
    <>
      <Head>
        <title>Profile</title>
        <meta name="description" content="Generated by create next app" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Header />
      <Profile user={name} image={image} />
      <Modal />
    </>
  )
}

export async function getServerSideProps(context) {
  const { query } = context
  const { name, image } = query
  return {
    props: {
      name: name || null,
      image: image || null,
    },
  }
}

export default profilePage
