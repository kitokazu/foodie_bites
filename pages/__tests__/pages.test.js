import React from 'react';
import { render, screen } from '@testing-library/react';
import "@testing-library/jest-dom";
import Header from '@/components/Header';
import homePage from '@/pages/index'
import profilePage from '@/pages/profilePage'

it('renders homepage unchanged', () => {
  const { container } = render(<homePage />)
  expect(container).toMatchSnapshot()
})