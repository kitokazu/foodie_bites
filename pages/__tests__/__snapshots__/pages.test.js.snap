// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <homepage />
</div>
`;

exports[`renders profile page unchanged 1`] = `
<div>
  <profilepage />
</div>
`;
